Sources/Fullscreen/AddressView/AddressView.swift

```diff
@@ -0,0 +1,222 @@
+//
+//  Copyright © FINN.no AS, Inc. All rights reserved.
+//
+
+import UIKit
+import MapKit
+
+public protocol AddressViewDelegate: class {
+    func addressViewDidSelectCopyButton(_ addressView: AddressView)
+    func addressViewDidSelectGetDirectionsButton(_ addressView: AddressView)
+    func addressViewDidSelectCenterMapButton(_ addressView: AddressView)
+    func addressView(_ addressView: AddressView, didSelectMapTypeAtIndex index: Int)
+}
+
+@objc public class AddressView: UIView {
+    public weak var delegate: AddressViewDelegate?
+
+    private lazy var mapTypeSegmentControl: UISegmentedControl = {
+        let control = UISegmentedControl(items: [])
+        control.translatesAutoresizingMaskIntoConstraints = false
+        control.addTarget(self, action: #selector(mapTypeChanged), for: .valueChanged)
+        control.tintColor = .primaryBlue
+        return control
+    }()
+
+    private lazy var segmentContainer: UIView = {
+        let segmentContainer = UIView()
+        segmentContainer.translatesAutoresizingMaskIntoConstraints = false
+        segmentContainer.backgroundColor = .white
+        segmentContainer.addSubview(mapTypeSegmentControl)
+        segmentContainer.layer.masksToBounds = false
+        segmentContainer.layer.shadowOpacity = 0.3
+        segmentContainer.layer.shadowRadius = 3
+        segmentContainer.layer.shadowOffset = .zero
+        segmentContainer.layer.shadowColor = UIColor.black.cgColor
+
+        if UIDevice.isIPad() {
+            NSLayoutConstraint.activate([
+                mapTypeSegmentControl.widthAnchor.constraint(equalToConstant: 350),
+                mapTypeSegmentControl.centerXAnchor.constraint(equalTo: segmentContainer.centerXAnchor)
+                ])
+        } else {
+            NSLayoutConstraint.activate([
+                mapTypeSegmentControl.leadingAnchor.constraint(equalTo: segmentContainer.leadingAnchor, constant: .mediumLargeSpacing),
+                mapTypeSegmentControl.trailingAnchor.constraint(equalTo: segmentContainer.trailingAnchor, constant: -.mediumLargeSpacing)
+                ])
+        }
+
+        NSLayoutConstraint.activate([
+            mapTypeSegmentControl.topAnchor.constraint(equalTo: segmentContainer.topAnchor, constant: .mediumLargeSpacing),
+            mapTypeSegmentControl.bottomAnchor.constraint(equalTo: segmentContainer.bottomAnchor, constant: -.mediumLargeSpacing)
+            ])
+
+        return segmentContainer
+    }()
+
+    private lazy var mapView: MKMapView = {
+        let view = MKMapView()
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.delegate = self
+        return view
+    }()
+
+    private lazy var addressCardView: AddressCardView = {
+        let view = AddressCardView()
+        view.delegate = self
+        view.translatesAutoresizingMaskIntoConstraints = false
```

Comment:
Same for this one, and `mapView` above, regarding `.init(withAutoLayout: true)`

Reply:
Updated all over the place. It seems like our components by default don't have this, like buttons and labels, I think it should be default since we do everything with AutoLayout. What do you think?

Reply:
Yup, agree. I don't think we have a single scenario where we use our `Button` or `Label` classes without setting `translatesAutoresizingMaskIntoConstraints` to `false`.